org.eclipse.january.examples/src/org/eclipse/january/examples/dataset/SlicingExamples.java

```diff
@@ -66,5 +71,26 @@ public void iterateImages2() throws DatasetException {
 		}
 	}
 
+	/**
+	 * Slice using basic Slice object
+	 * @throws Exception 
+	 */
+	@Test
+	public void iterateImagesND() throws DatasetException {
+		
+		final ILazyDataset lz = Random.lazyRand(64, 64, 100, 100);		
+		final PositionIterator it = new PositionIterator(new int[]{64, 64});
+		
+		while(it.hasNext()) {
+			int[] pos = it.getPos();
```

Comment:
primitive arrays pass by reference in Java:
	public int[] getPos() {
		return pos;
	}